### PR TITLE
Feature: Support pydantic v2

### PR DESCRIPTION
**Pull request by sudosubin**
### Related Issues

- Fixes https://github.com/yanyongyu/githubkit/issues/38

### Description

- Migrate pydantic v1 to v2
  - Updated `pyproject.toml`, `poetry.lock`
  - Updated deprecated functions/methods into v2 functions/methods (`codegen`, `githubkit`)
  - Actually I wanted to make githubkit support both pydantic v1 and v2, but it was too complex  and required so many changes
- Added `SetSchema`, because `List` with `unique_items` was deprecated, and recommended to use `Set`
- Updated to use `Annotated` for pydantic models
  - I only checked `Missing`, but there is a need to check if the same applies to `Union` required.

### CODE REVIEW

**Comment by yanyongyu** [comment]
I'm not sure that `set` is a good way to replace the unique list since the `set` type does not contain order information. But i find an other way to impl this in the pydantic issue https://github.com/pydantic/pydantic-core/pull/820#issuecomment-1670475909. If we don't need to keep the order information, the `set` type is recommended.

**Comment by sudosubin** [comment]
Thank you for the feedback. I applied your comment :D

GitHub's OpenAPI does not yet have a schema with `uniqueItems=True`, but if one exists in the future, it will be safer to keep the order.

**Comment by yanyongyu** [comment]
Thanks foryour great help with this feature. I still need to investigate how to make it compatible with pydantic v1 and v2 after this pr is merged. The codegen part should be rewrited to support features including compatibility layer, lazy loading logic and api versioning. This may take a long time before release since i do not have too much free time.

File: githubkit/graphql/models.py [hunk 1]
```diff
@@ -20,8 +20,9 @@ class GraphQLResponse(BaseModel):
     errors: Optional[List[GraphQLError]] = None
     extensions: Optional[Dict[str, Any]] = None
 
-    @root_validator
-    def validate_data_and_errors(cls, values: Dict[str, Any]):
+    @model_validator(mode="after")
+    @classmethod
+    def validate_data_and_errors(cls, values):
```
Comment (yanyongyu) [inline githubkit/graphql/models.py:25]:
```suggestion
    def validate_data_and_errors(cls, values: Dict[str, Any]):
```

File: pyproject.toml [hunk 2]
```diff
@@ -13,7 +13,7 @@ include = ["githubkit/py.typed"]
 
 [tool.poetry.dependencies]
 python = "^3.8"
-pydantic = "^1.9.1"
+pydantic = ">=2.0,<3.0.0"
```
Comment (yanyongyu) [inline pyproject.toml:16]:
```suggestion
pydantic = "^2.0.0"
```

**Review by yanyongyu — APPROVED** [review]

**Pull request merged**